scipy/signal/signaltools.py

```diff
@@ -3787,6 +3787,13 @@ def _validate_pad(padtype, padlen, x, axis, ntaps):
     return edge, ext
 
 
+def _validate_x(x):
+    x = np.asarray(x)
+    if x.ndim == 0:
+        raise ValueError('x must be at least 1D')
+    return x
+
+
 def sosfilt(sos, x, axis=-1, zi=None):
     """
     Filter data along one dimension using cascaded second-order sections.
```

Comment:
The text below should be edited, as this doesn't call `lfilter` anymore.

Reply:
Good catch, updated